src/schema/firefox-schemas-import.js

```diff
@@ -576,30 +578,18 @@ inner.isBrowserSchema = (_path) => {
   return schemaRegexes.some((re) => re.test(_path));
 };
 
-/**
- * Strip a null byte if it's the last character in a string.
- * */
-export function stripTrailingNullByte(str) {
-  if (str.indexOf('\u0000') === str.length - 1) {
-    return str.slice(0, -1);
-  }
-  return str;
-}
-
 export async function fetchSchemas({ inputPath, outputPath }) {
-  const tarball = fs.createReadStream(inputPath);
+  const openZip = util.promisify(yauzl.open);
+  const zipfile = await openZip(inputPath);
+  const openReadStream = util.promisify(zipfile.openReadStream.bind(zipfile));
+
   return new Promise((resolve, reject) => {
-    tarball
-      .pipe(new tar.Parse())
-      .on('entry', (entry) => {
-        // The updated tar library doesn't always strip null bytes from the
-        // end of strings anymore so we get to do that here.
-        const entryPath = stripTrailingNullByte(entry.path);
-        if (inner.isBrowserSchema(entryPath)) {
-          const filePath = path.join(outputPath, path.basename(entryPath));
-          entry.pipe(fs.createWriteStream(filePath));
-        } else {
-          entry.resume();
+    zipfile
```

Comment:
It looks that codecov.io is currently detecting a decreased coverage on this file, from a quick look it seems to me that it may be because we are missing to cover the "error" scenario (the one that trigger the rejection of this promise at line 595-597 of this file):
- https://codecov.io/gh/mozilla/addons-linter/pull/3036/changes#D1-596

It would be nice to cover it if it is not too much work (but I didn't looked if we can easily trigger it, e.g. if we can reasonably create some error condition, or if it would be easier to just make jest to mock the `yauzl` module and force such error condition using a fake `yauzl.open` method).

@EnTeQuAk would you mind to take a quick look into that?
(I'm fine with ignoring the codecov.io if it would require too much more time for this PR to be completed, there is a good chance that codecov.io is complaining just because we removed a good chunk of code from this file and the coverage percentage looks decreased even if it is not)

Reply:
I added 2f5a4eb4 for that, let's see if codecov likes it.

Reply:
thanks! codecov seems to be happy about it ;-)